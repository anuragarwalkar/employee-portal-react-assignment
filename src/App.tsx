import React from "react";
import { BrowserRouter, Redirect, Route, Switch } from "react-router-dom";
import Dashboard from "./containers/dashboard/dashboard";
import Layout from "./hoc/layout/layout";
import Auth from "./containers/auth/auth";
import { RootState } from "./models/rootState.model";
import { connect } from 'react-redux';
import styles from './App.module.css';

const App = ({isAuthenticated}: {isAuthenticated: boolean}) => {

  const routeGuard = (Component: any): any => {
    return isAuthenticated ? <Component /> : <Redirect to="/auth" />;
  };
  
  return (
    <BrowserRouter>
      <Layout className={styles.app}>
        <Switch>
          <Route path="/" exact render={() => routeGuard(Dashboard)} />
          <Route path="/auth" component={Auth}/>
        </Switch>
      </Layout>
    </BrowserRouter>
  );
}

const mapStateToProps = (state: RootState) => {
  const { token } = state.auth;

  return {
    isAuthenticated: token !== ''
  }
}

export default connect(mapStateToProps)(App);
