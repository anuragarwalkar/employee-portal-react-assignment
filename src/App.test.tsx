import React from "react";
import { App } from "./App";
import { configure, shallow, ShallowWrapper } from "enzyme";
import Enzyme from "enzyme-adapter-react-16";

configure({ adapter: new Enzyme() });

describe("<App />", () => {
  it("should render <App /> if user is not authenticated", () => {
    const wrapper: ShallowWrapper = shallow(<App isAuthenticated={false} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("should render <App /> if user is not authenticated", () => {
    const wrapper: ShallowWrapper = shallow(<App isAuthenticated={true} />);
    expect(wrapper).toMatchSnapshot();
  });
});
