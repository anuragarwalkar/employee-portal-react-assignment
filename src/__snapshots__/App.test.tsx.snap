// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should render <App /> if user is not authenticated 1`] = `ShallowWrapper {}`;

exports[`<App /> should render <App /> if user is not authenticated 2`] = `ShallowWrapper {}`;
